### PR TITLE
Emit To[ElementType]PtrOutput methods for go enum output types

### LINKED ISSUE #7496: [codegen/go] - Emit To[Type]PtrOutput methods for enum output types

### PR DESCRIPTION
**Pull request by komalali**
# Description

Fixes #7496 

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/codegen/go/gen.go [hunk 1]
```diff
@@ -782,6 +782,7 @@ func (pkg *pkgContext) genEnumType(w io.Writer, name string, enumType *schema.En
 	case "number":
 		goElementType = "float64"
 	}
+	asFuncName := Title(strings.Replace(elementType, "pulumi.", "", -1))
```
Comment (pgavlin) [inline pkg/codegen/go/gen.go:785]:
```suggestion
	asFuncName := strings.TrimPrefix(elementType, "pulumi.")
```

I don't think we need the call to `Title` here, do we? If we expect `elementType` to always be `pulumi.Foo`, then the result of trimming the prefix will always be exported. Or are there cases where we don't expect to see that prefix?

File: pkg/codegen/go/gen.go [hunk 2]
```diff
@@ -893,6 +915,20 @@ func (pkg *pkgContext) genEnumOutputTypes(w io.Writer, name string) {
...
+	fmt.Fprintf(w, "return o.To%sPtrOutputWithContext(context.Background())\n", asFuncName)
+	fmt.Fprint(w, "}\n\n")
+
+	fmt.Fprintf(w, "func (o %[1]sPtrOutput) To%[2]sPtrOutputWithContext(ctx context.Context) %[3]sPtrOutput {\n", name, asFuncName, elementType)
+	fmt.Fprintf(w, "return o.ApplyTWithContext(ctx, func(_ context.Context, e *%s) *%s {\n", name, goElementType)
+	fmt.Fprintf(w, "if e == nil {\n")
+	fmt.Fprintf(w, "return nil\n")
+	fmt.Fprintf(w, "}\n")
+	fmt.Fprintf(w, "s := %s(*e)\n", goElementType)
+	fmt.Fprintf(w, "return &s\n")
```
Comment (pgavlin) [inline pkg/codegen/go/gen.go:928]: tiny nit: 
```suggestion
	fmt.Fprintf(w, "v := %s(*e)\n", goElementType)
	fmt.Fprintf(w, "return &v\n")
```

File: pkg/codegen/go/gen.go [hunk 3]
```diff
@@ -878,6 +879,27 @@ func (pkg *pkgContext) genEnumOutputTypes(w io.Writer, name string) {
...
+	fmt.Fprint(w, "}\n\n")
+
+	fmt.Fprintf(w, "func (o %[1]sOutput) To%[2]sPtrOutput() %[3]sPtrOutput {\n", name, asFuncName, elementType)
+	fmt.Fprintf(w, "return o.To%sPtrOutputWithContext(context.Background())\n", asFuncName)
+	fmt.Fprint(w, "}\n\n")
+
+	fmt.Fprintf(w, "func (o %[1]sOutput) To%[2]sPtrOutputWithContext(ctx context.Context) %[3]sPtrOutput {\n", name, asFuncName, elementType)
+	fmt.Fprintf(w, "return o.ApplyTWithContext(ctx, func(_ context.Context, e %s) *%s {\n", name, goElementType)
+	fmt.Fprintf(w, "s := %s(e)\n", goElementType)
+	fmt.Fprintf(w, "return &s\n")
```
Comment (pgavlin) [inline pkg/codegen/go/gen.go:899]: tiny nit:
```suggestion
	fmt.Fprintf(w, "v := %s(e)\n", goElementType)
	fmt.Fprintf(w, "return &v\n")
```

**Review by pgavlin — APPROVED** [review]
Just a few nits; LGTM otherwise.

File: pkg/codegen/go/gen.go (hunk 1 above)
Comment (komalali) [reply pkg/codegen/go/gen.go:785]: You're probably right! This was just moving around some existing code.

**Pull request merged**